### PR TITLE
feat(oauth): Disable legacy signin/signup flows for OAuth

### LINKED ISSUE #592: Use email-first as default for oauth
blocked by #6528
blocked by #6529 
blocked by #6534

### LINKED ISSUE #2415: [fxa-content-server] Add email first test for OAuth WebChannels
image

From https://github.com/mozilla/fxa/pull/2320

### PR DESCRIPTION
**Pull request by shane-tomlinson**
fixes #592
fixes #2415

### CODE REVIEW

File: packages/fxa-content-server/tests/functional/oauth_permissions.js [hunk 1]
```diff
@@ -206,43 +209,20 @@ registerSuite('oauth permissions for untrusted reliers', {
...
-          .then(click(selectors.SIGNIN.SUBMIT_USE_SIGNED_IN))
+          .then(testElementExists(selectors.SIGNIN_PASSWORD.HEADER))
+          .then(click(selectors.SIGNIN_PASSWORD.SUBMIT_USE_SIGNED_IN))
 
           .then(testElementExists(selectors['123DONE'].AUTHENTICATED))
           .then(testUrlEquals(UNTRUSTED_OAUTH_APP))
       );
     },
 
-    'signin from signup page': function() {
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/tests/functional/oauth_permissions.js]: @philbooth  - at last, signin from signup is gone!

File: packages/fxa-content-server/tests/functional/oauth_query_param_validation.js [hunk 2]
```diff
@@ -368,21 +362,7 @@ registerSuite('oauth query parameter validation', {
             redirect_uri: TRUSTED_REDIRECT_URI,
             scope: TRUSTED_SCOPE,
           },
-          '#fxa-signup-header'
-        )
-      );
-    },
-
-    'authorization with signin (trusted)': function() {
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/tests/functional/oauth_query_param_validation.js]: This is a duplicate test with the oauth signin tests.

File: packages/fxa-content-server/tests/functional/oauth_query_param_validation.js [hunk 3]
```diff
@@ -417,41 +397,11 @@ registerSuite('oauth query parameter validation', {
...
               },
-              '#fxa-400-header'
+              selectors['400'].HEADER
             )
           )
       );
     },
-
-    'authorization with force_auth (trusted)': function() {
-      return this.remote.then(
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/tests/functional/oauth_query_param_validation.js]: also duplicates with the force_auth tests.

File: packages/fxa-content-server/tests/functional/oauth_sign_up.js [hunk 4]
```diff
@@ -223,84 +206,13 @@ registerSuite('oauth signup', {
       );
     },
 
-    'signup with existing account, coppa is valid': function() {
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/tests/functional/oauth_sign_up.js]: Again, no more signin from signup!

File: packages/fxa-content-server/tests/functional/oauth_email_first.js [hunk 5]
```diff
@@ -1,531 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-'use strict';
-
-const { registerSuite } = intern.getInterface('object');
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/tests/functional/oauth_email_first.js]: I migrated missing tests to oauth_sign_in.js

File: packages/fxa-content-server/tests/functional/oauth_choose_redirect.js [hunk 6]
```diff
@@ -1,58 +0,0 @@
...
-'use strict';
-
-const { registerSuite } = intern.getInterface('object');
-const TestHelpers = require('../lib/helpers');
-const FunctionalHelpers = require('./lib/helpers');
-const selectors = require('./lib/selectors');
-
-const config = intern._config;
-
-const CONTENT_SERVER_ROOT = config.fxaContentRoot;
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/tests/functional/oauth_choose_redirect.js]: oauth_choose_redirect is kinda the whole point of email-first

File: packages/fxa-content-server/tests/functional/oauth_permissions.js (hunk 1 above)
Comment (philbooth) [reply packages/fxa-content-server/tests/functional/oauth_permissions.js]: Finally, I can rest in peace. 😆

Comment (vladikoff) [reply packages/fxa-content-server/tests/functional/oauth_permissions.js]: :)

**Comment by shane-tomlinson** [comment]
@mozilla/fxa-devs - I'm calling this ready for review!

File: packages/fxa-content-server/app/scripts/models/auth_brokers/oauth-redirect.js [hunk 7]
```diff
@@ -67,6 +67,8 @@ export default BaseAuthenticationBroker.extend({
   }),
 
   defaultCapabilities: _.extend({}, proto.defaultCapabilities, {
+    disableLegacySigninSignup: true,
+    emailFirst: true,
```
Comment (dannycoates) [inline packages/fxa-content-server/app/scripts/models/auth_brokers/oauth-redirect.js]: 🎉

**Pull request merged**